core/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -60,7 +72,28 @@ protected WritePipelineResponse newResponse() {
 
     @Override
     protected void masterOperation(PutPipelineRequest request, ClusterState state, ActionListener<WritePipelineResponse> listener) throws Exception {
-        pipelineStore.put(clusterService, request, listener);
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+        nodesInfoRequest.clear();
+        nodesInfoRequest.ingest(true);
```

Comment:
scratch my comment here, since pipelines are available on all nodes, we need to check all nodes. For example the elected master node (which process the CRUD pipeline APIs) may not be an ingest node and simulate api can be executed on all nodes.

The ingest flag just controls whether ingest preprocessing can happen on a node.